lib/pbench/server/api/resources/datasets_compare.py

```diff
@@ -128,21 +128,27 @@ def _get(self, params: ApiParams, req: Request, context: ApiContext) -> Response
         stream_file = {}
         for dataset in datasets:
             try:
-                info = cache_m.get_inventory(dataset.resource_id, "result.csv")
-                file = info["stream"].read().decode("utf-8")
-                info["stream"].close()
+                file = cache_m.get_inventory_bytes(dataset.resource_id, "result.csv")
+            except CacheManagerError as e:
+                raise APIAbort(
+                    HTTPStatus.BAD_REQUEST,
+                    "unable to extract postprocessed data from {dataset.name}",
+                ) from e
```

Comment:
Are you sure that you want to report this message instead of the one in `e` like you do at line 70 in `lib/pbench/server/api/resources/datasets_inventory.py`?

Ditto line 73 in `lib/pbench/server/api/resources/datasets_visualize.py`.

Reply:
My logic was that `inventory` directly targets a specific file, and `NOT_FOUND` with a reference to the file seemed appropriate. Here and in visualize, we're asking to graph the benchmark summary, and the choice of file is implicit, so a more generic wording targeting the semantics (we can't find the postprocessed data) with `BAD_REQUEST` (you targeted an inapplicable dataset) seemed more appropriate.